homeassistant/components/hko/config_flow.py

```diff
@@ -0,0 +1,98 @@
+"""Config flow for Hong Kong Observatory integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from async_timeout import timeout
+from hko import HKO, LOCATIONS, HKOError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_LOCATION
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_LOCATION, DOMAIN
+
+
+def get_loc_name(item):
+    """Return an array of supported locations."""
+    return item["LOCATION"]
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_LOCATION, default=DEFAULT_LOCATION): vol.In(
+            list(map(get_loc_name, LOCATIONS))  # Select Location
+        )
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect."""
+
+    if data[CONF_LOCATION] not in list(map(get_loc_name, LOCATIONS)):
+        raise InvalidLocation
```

Comment:
There's no need to check this since the schema ensure it can't happen to my understanding?

Reply:
Correct